quantum/config_common.h

```diff
@@ -284,6 +284,86 @@
+#        define G5 PAL_LINE(GPIOG, 5)
+#        define G6 PAL_LINE(GPIOG, 6)
+#        define G7 PAL_LINE(GPIOG, 7)
+#        define G8 PAL_LINE(GPIOG, 8)
+#        define G9 PAL_LINE(GPIOG, 9)
+#        define G10 PAL_LINE(GPIOG, 10)
+#        define G11 PAL_LINE(GPIOG, 11)
+#        define G12 PAL_LINE(GPIOG, 12)
+#        define G13 PAL_LINE(GPIOG, 13)
+#        define G14 PAL_LINE(GPIOG, 14)
+#        define G15 PAL_LINE(GPIOG, 15)
+#        define H0 PAL_LINE(GPIOH, 0)
+#        define H1 PAL_LINE(GPIOH, 1)
+#        define H2 PAL_LINE(GPIOH, 2)
+#        define H3 PAL_LINE(GPIOH, 3)
+#        define H4 PAL_LINE(GPIOH, 4)
+#        define H5 PAL_LINE(GPIOH, 5)
+#        define H6 PAL_LINE(GPIOH, 6)
+#        define H7 PAL_LINE(GPIOH, 7)
+#        define H8 PAL_LINE(GPIOH, 8)
+#        define H9 PAL_LINE(GPIOH, 9)
+#        define H10 PAL_LINE(GPIOH, 10)
+#        define H11 PAL_LINE(GPIOH, 11)
+#        define H12 PAL_LINE(GPIOH, 12)
+#        define H13 PAL_LINE(GPIOH, 13)
+#        define H14 PAL_LINE(GPIOH, 14)
+#        define H15 PAL_LINE(GPIOH, 15)
+#        define I0 PAL_LINE(GPIOI, 0)
+#        define I1 PAL_LINE(GPIOI, 1)
+#        define I2 PAL_LINE(GPIOI, 2)
+#        define I3 PAL_LINE(GPIOI, 3)
+#        define I4 PAL_LINE(GPIOI, 4)
+#        define I5 PAL_LINE(GPIOI, 5)
+#        define I6 PAL_LINE(GPIOI, 6)
+#        define I7 PAL_LINE(GPIOI, 7)
+#        define I8 PAL_LINE(GPIOI, 8)
+#        define I9 PAL_LINE(GPIOI, 9)
+#        define I10 PAL_LINE(GPIOI, 10)
+#        define I11 PAL_LINE(GPIOI, 11)
+#        define I12 PAL_LINE(GPIOI, 12)
+#        define I13 PAL_LINE(GPIOI, 13)
+#        define I14 PAL_LINE(GPIOI, 14)
+#        define I15 PAL_LINE(GPIOI, 15)
+#        define J0 PAL_LINE(GPIOJ, 0)
+#        define J1 PAL_LINE(GPIOJ, 1)
+#        define J2 PAL_LINE(GPIOJ, 2)
+#        define J3 PAL_LINE(GPIOJ, 3)
+#        define J4 PAL_LINE(GPIOJ, 4)
+#        define J5 PAL_LINE(GPIOJ, 5)
+#        define J6 PAL_LINE(GPIOJ, 6)
+#        define J7 PAL_LINE(GPIOJ, 7)
+#        define J8 PAL_LINE(GPIOJ, 8)
+#        define J9 PAL_LINE(GPIOJ, 9)
+#        define J10 PAL_LINE(GPIOJ, 10)
+#        define J11 PAL_LINE(GPIOJ, 11)
+#        define J12 PAL_LINE(GPIOJ, 12)
+#        define J13 PAL_LINE(GPIOJ, 13)
+#        define J14 PAL_LINE(GPIOJ, 14)
+#        define J15 PAL_LINE(GPIOJ, 15)
+#        define K0 PAL_LINE(GPIOK, 0)
+#        define K1 PAL_LINE(GPIOK, 1)
+#        define K2 PAL_LINE(GPIOK, 2)
+#        define K3 PAL_LINE(GPIOK, 3)
+#        define K4 PAL_LINE(GPIOK, 4)
+#        define K5 PAL_LINE(GPIOK, 5)
+#        define K6 PAL_LINE(GPIOK, 6)
+#        define K7 PAL_LINE(GPIOK, 7)
+#        define K8 PAL_LINE(GPIOK, 8)
+#        define K9 PAL_LINE(GPIOK, 9)
+#        define K10 PAL_LINE(GPIOK, 10)
+#        define K11 PAL_LINE(GPIOK, 11)
+#        define K12 PAL_LINE(GPIOK, 12)
+#        define K13 PAL_LINE(GPIOK, 13)
+#        define K14 PAL_LINE(GPIOK, 14)
+#        define K15 PAL_LINE(GPIOK, 15)
```

Comment:
These might be a problem, unless we standardise on `kXX` for layout macros.

Reply:
Ohhh, good point. Hmmmm.

Reply:
I'll mark this PR as a draft while we discuss.

Reply:
well, don't we usually recommend/enforce lowercase for the matrix?  If not, it may be worth doing so. 
Also, good thing it doesn't hit L, because a bunch of splits use L##.

Reply:
I'm not sure I want to count on users remembering to stick to lowercase for layout macros. I'd prefer to change the letter or the notation system.

Reply:
That's a good point. Though, in theory, it should error out when trying to compile? 

Otherwise, I vote for `q`.

Reply:
I like the idea of using `q`; in spite of how common the letter Q is around the project, it's unlikely to be used or needed anywhere else.

Reply:
Quay!

Reply:
Added a guard around the GPIOK definitions -- `#define KEYBOARD_REQUIRES_GPIOK` -- which opts-in to defining them.